src/ui/Button/Link.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC, HTMLProps } from 'react';
+import NextLink, { LinkProps as NextLinkProps } from 'next/link';
+import {
+  outlineStyle,
+  ghostStyle,
+  secondaryStyle,
+  primaryStyle,
+  linkStyle,
+  BaseProps,
+} from './styles';
+
+type LinkProps = Omit<HTMLProps<HTMLAnchorElement>, keyof BaseProps> &
```

Comment:
This looks complex. Try this. 
```tsx
type Props = NextLinkProps | AnchorProps

const isNextLinkProps = (props: Props): Props is NextLinkProps => {return true;}

() => {
if(isNextLinkProps) return <NextLink />
return <a />
}
```

Reply:
The `Omit<HTMLProps<HTMLAnchorElement>, keyof BaseProps>` is to solve a werid TypeScript issue.

The `HTMLProps<HTMLAnchorElement>` (`AnchorProps`) type somehow has a `size` property in it which is not supposed to be there. When unioned with `BaseProps`, the 2 `size` properties conflict and result in the type `undefined`.

That's why I exclude all `keyof BaseProps` from the `AnchorProps` although they're tehcnically not there in the first place.

Reply:
And sometime we might need props in both NextLink and <a> so I don't think this would do...

Reply:
NextLinkProps = LinkProps & AnchorProps;

Reply:
I'll try that but ate you sure they don't have any overlapping props besides href?